wps-common/src/main/java/au/org/aodn/aws/wps/status/StatusHelper.java

```diff
@@ -1,20 +1,179 @@
+        failed.setExceptionReport(getExceptionReport(message, code));
+        return failed;
+    }
+
+
+
+    public static ExceptionReport getExceptionReport(String message, String code)
+    {
+        ExceptionReport report = new ExceptionReport();
+        ExceptionType type = new ExceptionType();
+        type.getExceptionText().add(message);
+        type.setExceptionCode(code);
+        report.getException().add(type);
+        return report;
+    }
+
+
+    public static String getExceptionReportString(String message, String code)
+    {
+        ExceptionReport report = getExceptionReport(message, code);
+        String responseDoc = null;
+
+        JAXBContext context;
+        try {
+            context = JAXBContext.newInstance(ExceptionReport.class);
+            Marshaller m = context.createMarshaller();
+            m.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
+            StringWriter writer = new StringWriter();
+            m.marshal(report, writer);
+            responseDoc = writer.toString();
+
+        } catch (JAXBException e) {
+            e.printStackTrace();
         }
+        return responseDoc;
+    }
+
+
+    public static ProcessStartedType getProcessStartedType(String message, Integer percentComplete)
+    {
+        ProcessStartedType started = new ProcessStartedType();
+        started.setValue(message);
+        started.setPercentCompleted(percentComplete);
+        return started;
+    }
+
+
+    public static final XMLGregorianCalendar getCreationDate()
+            throws DatatypeConfigurationException
+    {
+        GregorianCalendar currentTime = new GregorianCalendar();
+        currentTime.setTime(new Date());
+        currentTime.setTimeZone(TimeZone.getTimeZone("GMT"));
+        return DatatypeFactory.newInstance().newXMLGregorianCalendar(currentTime);
+    }
 
-        return statusDocument;
+
+    public static void addExecuteOutputReference(ExecuteResponse response, String outputIdentifier, String outputHref)
+    {
+        net.opengis.wps._1_0.ObjectFactory objectFactory = new net.opengis.wps._1_0.ObjectFactory();
+        OutputDataType output = objectFactory.createOutputDataType();
+
+        OutputReferenceType outputReference = objectFactory.createOutputReferenceType();
+        outputReference.setHref(outputHref);
+        output.setReference(outputReference);
+
+        CodeType outputIdentifierCode = new CodeType();
+        outputIdentifierCode.setValue(outputIdentifier);
+        output.setIdentifier(outputIdentifierCode);
+        if(response.getProcessOutputs() == null) {
+            ExecuteResponse.ProcessOutputs outputs = objectFactory.createExecuteResponseProcessOutputs();
+            response.setProcessOutputs(outputs);
+        }
+        response.getProcessOutputs().getOutput().add(output);
+    }
+
+    public static void addExecuteLiteralProcessOutput(ExecuteResponse response, String outputIdentifier, InputStream inStream)
```

Comment:
Not used anywhere. Would it be better to remove this until we need to support LiteralProcessOutputs?

Reply:
Possibly  :)  It may be useful to someone some day though  :)